app/src/main/java/fr/neamar/kiss/result/ShortcutsResult.java

```diff
@@ -132,7 +139,30 @@ public View display(final Context context, View view, @NonNull ViewGroup parent,
     }
 
     public Drawable getDrawable(Context context) {
-        return new BitmapDrawable(context.getResources(), shortcutPojo.getIcon(context));
+        Drawable shortcutDrawable = null;
+        if(android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            final LauncherApps launcherApps = (LauncherApps) context.getSystemService(Context.LAUNCHER_APPS_SERVICE);
+            assert launcherApps != null;
+
+            if (launcherApps.hasShortcutHostPermission()) {
+                LauncherApps.ShortcutQuery query = new LauncherApps.ShortcutQuery();
+                query.setPackage(shortcutPojo.packageName);
+                query.setShortcutIds(Collections.singletonList(shortcutPojo.getOreoId()));
+                query.setQueryFlags(FLAG_MATCH_DYNAMIC | FLAG_MATCH_MANIFEST | FLAG_MATCH_PINNED);
+
+                List<UserHandle> userHandles = launcherApps.getProfiles();
+
+                // Find the correct UserHandle, and retrieve the icon.
+                for (UserHandle userHandle : userHandles) {
+                    List<ShortcutInfo> shortcuts = launcherApps.getShortcuts(query, userHandle);
+                    if (shortcuts != null && shortcuts.size() > 0) {
+                        shortcutDrawable = launcherApps.getShortcutIconDrawable(shortcuts.get(0), 0);
+                    }
+                }
+            }
+        }
+
+        return shortcutDrawable;
```

Comment:
Wouldn't this return null for devices < Oreo?

Reply:
Yes it would.
In the original code, I did not find any way a pre-Oreo device could get a shortcut icon: in fact, the way the icon is retrieved here is more or less the same as in the original code, and it only works on Oreo+
Also, the case where the sortcutDrawable would be null was already handled, by putting the app icon instead and a small stock android "send" icon